org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMAttributeFormatter.java

```diff
@@ -78,7 +67,47 @@ public void formatAttribute(DOMAttr attr, int prevOffset, boolean singleAttribut
 				int attrValueStart = attr.getNodeAttrValue().getStart(); // <foo attr = |""
 				removeLeftSpaces(delimiterOffset, attrValueStart, edits);
 			}
-			formatAttributeValue(attr, parentConstraints.getIndentLevel(), edits);
+
+			// Compute max line width for attribute name and indents if maxLineWidth is
+			// enabled
+			if (isMaxLineWidthSupported()) {
+				int availableLineWidth = parentConstraints.getAvailableLineWidth();
+				if (isPreserveAttributeLineBreaks() && hasLineBreak(prevOffset, attr.getStart())) {
+					availableLineWidth = getMaxLineWidth() - getTabSize() * (indentLevel + 1);
+				} else if (isSplitAttributes() && !singleAttribute) {
+					availableLineWidth = getMaxLineWidth()
+							- getTabSize() * (indentLevel + getSplitAttributesIndentSize());
+				} else {
+					// counts the space between the start tag name and attribute value
+					availableLineWidth--;
+				}
+				// Add width for length of attribute name and 3 for '=""'
+				// between start tag name and tag
+				attributeNamelength = attrNameEnd - attr.getNodeAttrName().getStart() + 3;
+				parentConstraints.setAvailableLineWidth(availableLineWidth - attributeNamelength);
+			}
+			formatAttributeValue(attr, parentConstraints, edits);
+		}
+
+		if (!alreadyIndented) {
+			int from = prevOffset;
+			int to = attr.getStart();
+			if (isMaxLineWidthSupported() && parentConstraints.getAvailableLineWidth() < 0
+					&& !isSplitAttributes()) {
+				replaceLeftSpacesWithIndentation(indentLevel + 1, from, to, true, edits);
+				parentConstraints.setAvailableLineWidth(
+						getMaxLineWidth() - getTabSize() * (indentLevel + 1) - attributeNamelength
+								- attr.getValue().length());

```

Comment:
I wonder if you take care of this usecase with attribute:

 * attr=
 * attr="
 * attr         =
 * attr         =        "
 *  attr         =        "    "

Reply:
I suggest that you write test with invalid attribute like I described below to check that we have not some NPE (I fear that attr.getValue() can return null in some invalid usecase attribute) and I have the impression that you don't take care of spaces beteween delimiter (=), but perhaps I'm wrong.